### PR TITLE
polish imperative codes

### PR DESCRIPTION
**Pull request by panyx0718**
test=develop

### CODE REVIEW

File: paddle/fluid/imperative/layer.cc [hunk 1]
```diff
@@ -131,8 +134,9 @@ std::map<std::string, std::vector<VarBase*>> OpBase::ApplyGrad() {
   std::map<std::string, std::vector<framework::Variable*>> grad_outputs;
   if (backward_id_ > 0) {
     VLOG(3) << "py_layer_grad";
-    grad_outputs["Out@GRAD"] =
-        PyLayer::ApplyGrad(backward_id_, grad_input_vars_["X@GRAD"]);
+    grad_outputs[framework::GradVarName(PyLayer::kFwdOut)] = PyLayer::ApplyGrad(
```
Comment (velconia) [inline paddle/fluid/imperative/layer.cc:137]: How about a static GradVarName

File: paddle/fluid/imperative/tracer.cc [hunk 2]
```diff
@@ -164,28 +164,30 @@ std::vector<VarBase*> Tracer::PyTrace(OpBase* op,
...
     out->pre_op_ = op;
-    out->pre_op_out_name_ = "Out";
+    out->pre_op_out_name_ = PyLayer::kFwdOut;
     out->pre_op_out_idx_ = i;
   }
   if (!stop_gradient) {
-    auto& grad_input_vars = op->grad_input_vars_["X@GRAD"];
-    auto& grad_output_vars = op->grad_output_vars_["Out@GRAD"];
+    auto& grad_input_vars =
+        op->grad_input_vars_[framework::GradVarName(PyLayer::kFwdInp)];
```
Comment (velconia) [inline paddle/fluid/imperative/tracer.cc:188]: same above

File: python/paddle/fluid/imperative/layers.py [hunk 3]
```diff
@@ -54,6 +54,25 @@ class PyLayer(core.PyLayer):
...
+    def _do_forward(cls, inputs):
+        return cls._to_tuple(cls.forward(inputs))
+
+    @classmethod
+    def _do_backward(cls, inputs):
+        return cls._to_tuple(cls.backward(inputs))
+
+    @staticmethod
+    def _to_tuple(inputs):
+        if not isinstance(inputs, list) and not isinstance(inputs, tuple):
```
Comment (velconia) [inline python/paddle/fluid/imperative/layers.py:67]: How about cast the return value to tuple in C++ part, in the help of pybind11

File: paddle/fluid/imperative/layer.cc (hunk 1 above)
Comment (panyx0718) [reply paddle/fluid/imperative/layer.cc:137]: What does "static GradVarName" mean? Have a string call X@GRAD?

File: paddle/fluid/imperative/tracer.cc (hunk 2 above)
Comment (panyx0718) [reply paddle/fluid/imperative/tracer.cc:188]: same above

File: python/paddle/fluid/imperative/layers.py (hunk 3 above)
Comment (panyx0718) [reply python/paddle/fluid/imperative/layers.py:67]: SG. probably not in this PR.

**Review by velconia — APPROVED** [review]
LGTM

**Pull request merged**